src/ewatercycle/_forcings/penman_monteith.py

```diff
@@ -0,0 +1,60 @@
+from pathlib import Path
+from ewatercycle.base.forcing import (
+    DefaultForcing,
+    DistributedUserForcing,
+    LumpedUserForcing,
+)
+from ewatercycle.esmvaltool.schema import Dataset
+
+
+class PenmanMonteith(DefaultForcing):
+    """Forcing object with variables for computing the Penman-Monteith equation.
+
+    Contains the following variables:
+        pr - precipitation
+        tas - near-surface air temperature
+        hurs - near-surface relative humidity
+        sfcWind - near-surface wind speed
+        rsds - near-surface downwelling shortwave radiation
+        rsus - near-surface upwelling shortwave radiation
+        rlds - near-surface downwelling longwave radiation
+        rlus - near-surface upwelling longwave radiation
+
+    Which together with the following local parameters is sufficient to solve the
+    Penman-Monteith equation (ignoring ground heat flux, FAO 56).
+
+    Do note that the "near-surface" height in weather and climate models is usually
+    10 m. The exact surface height of each variable can be retrieved from the variable's
+    attributes, if this information was present in the CMORized data.
+
+    FAO 56: https://www.fao.org/3/x0490e/x0490e06.htm
+    """
+
+    @classmethod
+    def generate(  # type: ignore[override]
+        cls: type["PenmanMonteith"],
+        dataset: str | Dataset | dict,
+        start_time: str,
+        end_time: str,
+        shape: str | Path,
+        directory: str | None = None,
+        **model_specific_options,
+    ) -> "PenmanMonteith":
+        return super().generate(
+            dataset,
+            start_time,
+            end_time,
+            shape,
+            directory,
+            variables=("pr", "tas", "hurs", "sfcWind", "rsds", "rlds", "rsus", "rlus"),
+            postprocessor=None,
+            **model_specific_options,
+        )
+
+
+class LumpedPenmanMonteithForcing(PenmanMonteith, LumpedUserForcing):  # type: ignore[misc]
+    ...
+
+
+class DistributedPenmanMonteithForcing(PenmanMonteith, DistributedUserForcing):  # type: ignore[misc]
```

Comment:
I tried to run 

```
pmforcing = ewatercycle.forcing.sources.LumpedPenmanMonteithForcing.generate(
     dataset='ERA5',
    start_time="2000-01-01T00:00:00Z",
    end_time="2001-01-01T00:00:00Z",
    shape=shape.absolute(),
)   
print(pmforcing)
```

But got error:
ERROR:esmvalcore._recipe.recipe:Missing data for preprocessor diagnostic/rsus:
- Missing data for Dataset: rsus, day, OBS6, ERA5, 1
ERROR:esmvalcore._recipe.recipe:Missing data for preprocessor diagnostic/rlus:
- Missing data for Dataset: rlus, day, OBS6, ERA5, 1

This is with dcache mounted, to get the generate to work, those vars should be added to dcache.

If I try with cmip_dataset from user guide I get `OSError: [Errno 113] No route to host` error while LumpedMakkinkForcing worked this morning.

Reply:
I had another good look, and turns out that there's not too much overlap between the CMIP and ERA5 variables :roll_eyes: 

With some postprocessors I now have it working for both ERA5 and CMIP6.

> If I try with cmip_dataset from user guide I get OSError: [Errno 113] No route to host error while LumpedMakkinkForcing worked this morning.

I think ESGF had some issues. Should be working now (it runs fine for me now).